### ISSUE TITLE
Evaluate expressions from within an action

### ISSUE THREAD

**Issue by thisismydesign** [actions/toolkit#472]
**Describe the enhancement**

I'm writing an action. Would like to call `${{ hashFiles(...) }}` and `${{ runner.OS }}` and get the result from within the action. Is this possible? I understand I could have these as inputs, however the purpose of my action is to simplify that process.

**Code Snippet**

Something along these lines:

`index.js`
```js
const core = require("@actions/core");

async function run() {
  const hash = core.runExpression('hashFiles(...)')
}

run();
```

**Comment by thisismydesign** [comment]
Related to https://github.com/actions/toolkit/issues/430

**Comment by ericsciple** [comment]
You can use the env var `RUNNER_OS`

The hashFiles source code [is here](https://github.com/actions/runner/blob/master/src/Misc/expressionFunc/hashFiles/src/hashFiles.ts)

@joshmgross @TingluoHuang should we add hashFiles to the toolkit?

@thisismydesign today the runner evaluates all expressions. Currently not consumable from javascript action. You can also add a default input with an expression, for example actions/checkout uses an expression to set the [default repository input](https://github.com/actions/checkout/blob/2ff2fbdea48a8f5da77a31e7dd5ecb46c017ffc3/action.yml#L6).

**Comment by thisismydesign** [comment]
Thanks! Would be great for my use case to have access to `hashFiles`.

**Comment by joshmgross** [comment]
>should we add hashFiles to the toolkit?

Absolutely. With `@actions/cache` now available as an npm package, action authors should have a way to consume `hashFiles` without requiring user input.

**Comment by thisismydesign** [comment]
Awesome, that's exactly my use case :)

**Comment by ericsciple** [comment]
Updated the title.

@thisismydesign as a workaround you can copy the relevant source code from the link i pasted above. It's just that one file, the only dependency is `@actions/glob` and all other dependencies are from provided by nodejs.

**Comment by eine** [comment]
Since I'm writing a JS Action, I'm using [folder-hash](https://www.npmjs.com/package/folder-hash) instead of copying [hashFiles.ts](https://github.com/actions/runner/blob/master/src/Misc/expressionFunc/hashFiles/src/hashFiles.ts). Nonetheless, I'd prefer to have the function available in this toolkit.

Ref eine/setup-msys2#23
